### PR TITLE
Enhancement : added pdf downloading feature

### LINKED ISSUE #3686: Enhancement : PDF downloader
### Is your proposal related to a problem?
Currently, we are not having any option to download a pdf file

### Describe the solution you'd like
Adding download button to download pdf

### If the feature is approved, would you be willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by manishkushare**
Resolve #3686

### CODE REVIEW

**Comment by shubh22** [comment]
Task linked: [CU-2kbtjrt Pdf download ](https://app.clickup.com/t/2kbtjrt)

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@manishkushare As discussed, check with Jaseem for the UI changes and update the tooltip

**Comment by manishkushare** [comment]
@kavinvenkatachalam , made the required UI changes, please take a look at it, whenever possible.

**Comment by kavinvenkatachalam** [comment]
@manishkushare Just a small change, If I toggle off both `Download` and `Page Control`, I can see the footer. In this case, we can hide the complete footer area. 

image

**Comment by manishkushare** [comment]
@kavinvenkatachalam , made the requested changes, please check it, whenever possible.

File: frontend/src/Editor/Components/PDF.jsx [hunk 1]
```diff
@@ -47,46 +51,107 @@ export const PDF = React.memo(({ styles, properties, width, height }) => {
...
+          <Page
+            pageNumber={index + 1}
+            width={scale ? width - 12 : undefined}
+            height={scale ? undefined : height}
+            key={`page_${index + 1}`}
+            inputRef={(el) => (pageRef.current[index] = el)}
+          />
+        ))}
+      </Document>
+    </>
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Components/PDF.jsx:84]:
```suggestion
      <Document
        file={url}
        onLoadSuccess={onDocumentLoadSuccess}
        onLoadError={onDocumentLoadError}
        className="pdf-document"
      >
        {Array.from(new Array(numPages), (el, index) => (
          <Page
            pageNumber={index + 1}
            width={scale ? width - 12 : undefined}
            height={scale ? undefined : height}
            key={`page_${index + 1}`}
            inputRef={(el) => (pageRef.current[index] = el)}
          />
        ))}
      </Document>
```

We can remove the fragments as we already have a common parent

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@manishkushare When I change the PDF URL, I can see the footer displayed while loading the new PDF. Can we hide that?

**Comment by manishkushare** [comment]
@kavinvenkatachalam , made the requested changes, please take a look at it, whenever possible

**Review by kavinvenkatachalam — APPROVED** [review]

File: frontend/src/Editor/Components/PDF.jsx [hunk 2]
```diff
@@ -58,35 +84,75 @@ export const PDF = React.memo(({ styles, properties, width, height }) => {
...
+                    ›
+                  </button>
+                </div>
+              </>
+            )}
+            {showDownloadOption && (
+              <div
+                className="download-icon-outer-wrapper text-dark"
+                style={downloadIconOuterWrapperStyles}
+                onClick={() => downloadFile(url, pdfName)}
```
Comment (gondar00) [inline frontend/src/Editor/Components/PDF.jsx:143]: Would it be possible to use `<PDFDownloadLink />` here?
ref: https://react-pdf.org/advanced#on-the-fly-rendering

Comment (kavinvenkatachalam) [reply frontend/src/Editor/Components/PDF.jsx:143]: @gondar00 We need to install a `react-pdf/renderer` package to make use of it. So, we avoided that

Comment (gondar00) [reply frontend/src/Editor/Components/PDF.jsx:143]: On a closer look @ [`<PDFDownloadLink />`](https://github.com/diegomura/react-pdf/blob/d011983204cf45876594fa361f24b47e86c612c9/packages/renderer/src/dom/PDFDownloadLink.js) it seems like a wrapper around a anchor tag only, can ignore it

**Pull request merged**